apluslms_roman/tests/test_cli.py

```diff
@@ -0,0 +1,278 @@
+        out, _ = command_test(self, "roman -f temp.yml step list")
+        self.assertEqual("The project config has no steps.", out)
+
+
+class TestStepAddAction(TestCase):
+
+    @use_temp_file
+    def test_step_add(self):
+        out, _ = command_test(self, (
+            "roman -f temp.yml step add hello-world --name new_step".split() +
+            ["--cmd", "\"make touchrst html\""] +
+            "--env test1=a test2=${test1} --mnt /compile".split()
+        ))
+        self.assertEqual("Step successfully added to config.", out)
+        out, _ = command_test(self, "roman -f temp.yml step list")
+        self.assertIn("new_step", out)
+
+    @use_temp_file
+    def test_step_add_invalid_env(self):
+        _, err = command_test(self, "roman -f temp.yml step add hello-world --env a: b", 1)
+        self.assertEqual(
+            "env is a dict, so values need to be in key=val format, e.g. a=1 b=2", err)
+
+    @use_temp_file
+    @add_hello
+    def test_step_add_duplicate_name(self):
+        _, err = command_test(self, "roman -f temp.yml step add hello-world --name hello", 1)
+        self.assertEqual("A step with the name 'hello' already exists.", err)
+
+    @use_temp_file
+    def test_step_add_validation_error(self):
+        command_test(self, "roman -f temp.yml step add hello-world --name 123", 1)
+
+
+class TestStepRemoveAction(TestCase):
+
+    @mock.patch('apluslms_roman.cli.input', create=True)
+    @use_temp_file
+    @add_hello
+    def test_rm_action_confirm_no(self, mocked_input):
+        mocked_input.side_effect = ['n']
+        command_test(self, "roman -f temp.yml step rm hello")
+        out, _ = command_test(self, "roman -f temp.yml step list")
+        self.assertIn("hello", out)
+
+    @mock.patch('apluslms_roman.cli.input', create=True)
+    @use_temp_file
+    @add_hello
+    def test_rm_action_confirm_yes(self, mocked_input):
+        mocked_input.side_effect = ['y']
+        command_test(self, "roman -f temp.yml step rm hello")
+        out, _ = command_test(self, "roman -f temp.yml step list")
+        self.assertEqual("The project config has no steps.", out)
+
+    @mock.patch('apluslms_roman.cli.input', create=True)
+    @use_temp_file
+    @add_hello
+    def test_rm_action_force(self, mocked_input):
+        mocked_input.side_effect = ['n']
+        command_test(self, "roman -f temp.yml step rm -f hello")
+        out, _ = command_test(self, "roman -f temp.yml step list")
+        self.assertEqual("The project config has no steps.", out)
+
+    @use_temp_file
+    @add_hello
+    def test_rm_action_index_error(self):
+        _, err = command_test(self, "roman -f temp.yml step rm -f 2", 1)
+        self.assertIn("Index is out of range", err)
+
+    @use_temp_file
+    @add_hello
+    def test_rm_action_key_error(self):
+        _, err = command_test(self, "roman -f temp.yml step rm -f hei", 1)
+        self.assertEqual("There is no step called 'hei'", err)
+
+chdir(COURSE)
```

Comment:
We shouldn't change current dir, as that might do unintended side effects. A bit better is to do someting like: `command_test(self, "roman -f '%s' step rm -f hei" % (join(COURSE, 'temp.yml'), 1)`

Btw. I would implement a parent class like
```
class Foo(unitetest.TestCase):
  def command_test(self):
    pass # implementation here

class Test1(Foo):
  def test_rm_action_key_error(self):
    _, err = self.command_test("roman -f plaa")
```

`command_test` could even have pattern `def command_test(config_filename, subcommand)` and would then do something like `cmd = ['roman', '-f', config_filename] + shlex.split(subcommand)`

Reply:
Yeah that would look a lot better... Still not very used to doing class based programming. :p